src/components/auth/SignUp.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { signUp } from '../../redux/actions/authAuctions';
+import { Redirect } from 'react-router-dom';
+import { Link } from 'react-router-dom';
+import '../../css/auth.css';
+
+class SignUp extends Component {
+  state = {
+    email: '',
+    password: '',
+    firstName: '',
+    lastName: ''
+  };
+  onInputChange = e => {
+    this.setState({ [e.target.id]: e.target.value });
+  };
+
+  onFormSubmit = e => {
+    e.preventDefault();
+    this.props.signUp(this.state);
+  };
+
+  render() {
+    const { authError, auth } = this.props;
+    if (auth.uid) return <Redirect to="/" />;
```

Comment:
Do this via another `route protection hoc` which does the opposite: if the user is logged in, redirects to index.